server/src/main/java/org/elasticsearch/script/MultiVectorScoreScriptUtils.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.script;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.script.field.vectors.DenseVector;
+import org.elasticsearch.script.field.vectors.MultiDenseVectorDocValuesField;
+
+import java.io.IOException;
+import java.util.HexFormat;
+import java.util.List;
+
+public class MultiVectorScoreScriptUtils {
+
+    public static class MultiDenseVectorFunction {
+        protected final ScoreScript scoreScript;
+        protected final MultiDenseVectorDocValuesField field;
+
+        public MultiDenseVectorFunction(ScoreScript scoreScript, MultiDenseVectorDocValuesField field) {
+            this.scoreScript = scoreScript;
+            this.field = field;
+        }
+
+        void setNextVector() {
+            try {
+                field.setNextDocId(scoreScript._getDocId());
+            } catch (IOException e) {
+                throw ExceptionsHelper.convertToElastic(e);
+            }
+            if (field.isEmpty()) {
+                throw new IllegalArgumentException("A document doesn't have a value for a multi-vector field!");
+            }
+        }
+    }
+
+    public static class ByteMultiDenseVectorFunction extends MultiDenseVectorFunction {
+        protected final byte[][] queryVector;
+
+        /**
+         * Constructs a dense vector function used for byte-sized vectors.
+         *
+         * @param scoreScript The script in which this function was referenced.
+         * @param field The vector field.
+         * @param queryVector The query vector.
+         */
+        public ByteMultiDenseVectorFunction(ScoreScript scoreScript, MultiDenseVectorDocValuesField field, List<List<Number>> queryVector) {
+            super(scoreScript, field);
+            if (queryVector.isEmpty()) {
+                throw new IllegalArgumentException("The query vector is empty.");
+            }
+            field.getElementType().checkDimensions(field.get().getDims(), queryVector.get(0).size());
+            this.queryVector = new byte[queryVector.size()][queryVector.get(0).size()];
+            float[] validateValues = new float[queryVector.size()];
+            int lastSize = -1;
+            for (int i = 0; i < queryVector.size(); i++) {
+                if (lastSize != -1 && lastSize != queryVector.get(i).size()) {
+                    throw new IllegalArgumentException("The query vector has an inconsistent number of dimensions.");
```

Comment:
```suggestion
                    throw new IllegalArgumentException("The query vector contains inner vectors which are have inconsistent dimensions.");
```
Might be more helpful to explain that the error is referring to the second dimension. I don't know if "inner" makes sense or saying "the second dimension". I wanted to say "term vectors" but that won't always be the case.